addons/point_of_sale/static/src/app/screens/receipt_screen/receipt/receipt_header/receipt_header.xml

```diff
@@ -11,7 +11,11 @@
                 <div>Tel:<t t-esc="props.data.company.phone" /></div>
             </t>
             <t t-if="props.data.company.vat">
-                <div><t t-esc="props.data.company.country?.vat_label || 'Tax ID'"/>: <t t-esc="props.data.company.vat" /></div>
+                <div>
+                    <t t-if="props.data.company.country_id?.vat_label" t-esc="props.data.company.country_id?.vat_label"/>
+                    <t t-else="">Tax ID</t>
+                    : <t t-esc="props.data.company.vat"/>
```

Comment:
You may want to move this to a getter on the component, so that `:` can be part of the translation.
```js
get vatText() {
    if (this.props.data.company.country_id?.vat_label) {
        return _t("%s(vatLabel): %(vat)s", {
            vatLabel: this.props.data.company.country_id?.vat_label,
            vat: this.props.data.company.vat,
        });
    }
    return _t("Tax ID: %(vat)s", { vat: this.props.data.company.vat });
}
```
For example, in French, a space should be inserted before the colon. Some other languages may have similar needs